### PR TITLE
Sortable objectives

### LINKED ISSUE #2728: Sortable objectives
add controls for explicitly sorting objectives within their context (session/course/programyear) via a drag-and-drop interface.

similar to the course/session learning materials sort manager.

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #2728

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit 2b113b319e6a11882f4ecc293841aa10dcaa552d

https://deploy-preview-2729--ilios-frontend.netlify.com

**Comment by saschaben** [comment]
Although everything seems good, the naked PR here does have a few intermittent issues. Let's make sure to review this: https://deploy-preview-2729--ilios-frontend.netlify.com/events/S0120170321O88893 currently causes a failure with ``` Cannot read property 'toString' of undefined

TypeError: Cannot read property 'toString' of undefined
    at https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:85:13045
    at Array.map (native)
    at i.<anonymous> (https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:85:13014)
    at c.p.get (https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:12:15548)
    at Object.n [as get] (https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:13:27990)
    at t.compute (https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:11:28135)
    at t.value (https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:11:26840)
    at t.compute (https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:11:28617)
    at t.value (https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:11:26840)
    at t.e.value (https://deploy-preview-2729--ilios-frontend.netlify.com/assets/vendor-eac877b62c8f66631370ca1efc9d9513.js:22:31489) ```

**Comment by stopfstedt** [comment]
@saschaben noted. i checked in on that error, and it comes out of the ilios-calendar addon. there's a corresponding PR in that project's repo out there.
i suppose the calendar addon needs to be tagged and pushed out first, then the dependency in frontend proper needs to be updated, and _then_ this will work.

**Comment by stopfstedt** [comment]
@saschaben i fixed the problem with session objectives borking the event details. pls have another look on  netlify, thanks!

File: tests/unit/mixins/sortable-objective-list-test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import Ember from 'ember';
+import SortableObjectiveListMixin from 'ilios/mixins/sortable-objective-list';
+import { module, test } from 'qunit';
+
+module('Unit | Mixin | sortable objective list');
+
+// Replace this with your real tests.
```
Comment (jrjohnson) [inline tests/unit/mixins/sortable-objective-list-test.js]: Isn't this a real test?

**Comment by jrjohnson** [comment]
Needs #2813 for the calendar stuff.

**Comment by jrjohnson** [comment]
@saschaben this is now ready for final review.  Sorting should eve be reflected correctly in the calendar.

**Comment by saschaben** [comment]
functionally AOK. let's spruce up the calendar display a bit: sort order objective display needs bullets, and the competency text (currently in smaller font following) should get a bold as well.

a future iteration will have to give better context for the buttons, but we can deal with that post release.

**Comment by stopfstedt** [comment]
@saschaben - thanks for the feedback. i opened up #illios/calendar#172 so we can track your requested changes as an enhancement request.

**Comment by saschaben** [comment]
👍

**Pull request merged**